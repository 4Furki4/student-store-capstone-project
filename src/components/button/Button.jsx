export default function Button({ buttonStyle, type, text, handleClick }) {
    const signInBtnSignInPage =
        "h-10 w-40 m-1 p-1 px-4 rounded-lg border-2 bg-pumpkin font-extralight text-[15px] text-white shadow-md";
    const forgotPasswordBtn =
        "h-10 w-40 m-1 p-1 px-2 rounded-lg border-2 bg-white font-extralight text-[15px] text-pumpkin shadow-md";
    const signUpBtnSignInPage =
        "h-9 w-36 m-1 p-1 rounded-lg border-2 bg-pumpkin font-extralight text-white text-sm shadow-sm";
    const signInBtnSignUpPage =
        "h-9 w-36 m-1 p-1 rounded-lg border-2 bg-darkPurple font-extralight text-sm text-white shadow-sm";
    const signUpBtnSignUpPage =
        "h-10 w-40 m-1 p-1 rounded-lg border-2 bg-darkPurple font-extralight text-[15px] text-white shadow-sm";
    const listItemBtn =
        "h-12 w-72 md:h-14 md:w-80 lg:h-14 lg:w-96 m-1 p-1 rounded-lg border-2 bg-pumpkin font-md text-[17px] text-white shadow-sm";
    const saveChangesBtn =
        "h-12 w-48 lg:h-14 lg:w-64 m-1 p-1 px-2 hover:bg-purple hover:shadow-lg rounded-lg border-2 bg-pumpkin font-md text-[17px] text-white shadow-sm";
    const allButtons = {
        orangeSignIn: signInBtnSignInPage,
        forgotPassword: forgotPasswordBtn,
        orangeSignUp: signUpBtnSignInPage,
        purpleSignIn: signInBtnSignUpPage,
        purpleSignUp: signUpBtnSignUpPage,
        listItem: listItemBtn,
        saveChanges: saveChangesBtn,
    };
    return (
        <div className='p-1'>
            <button
                className={allButtons[buttonStyle]}
                onClick={handleClick}
                type={type}
            >
                {text}
            </button>
        </div>
    );
}
