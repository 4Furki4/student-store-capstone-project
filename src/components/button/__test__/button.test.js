import renderer from "react-test-renderer";

import Button from "../Button";

it("renders correctly", () => {
    const tree = renderer
        .create(<Button buttonStyle='orangeSignIn' />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
