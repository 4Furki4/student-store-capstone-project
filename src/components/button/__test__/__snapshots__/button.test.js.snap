// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="p-1"
>
  <button
    className="h-10 w-40 m-1 p-1 px-4 rounded-lg border-2 bg-pumpkin font-extralight text-[15px] text-white shadow-md"
  />
</div>
`;
